.ci/go-static-checks.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+#  This file is part of cc-oci-runtime.
+#
+#  Copyright (C) 2017 Intel Corporation
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+set -e
+
+# May not be necessary for CI machine runs, but enables local runs for some
+# users
+PATH=${PATH}:${GOPATH}/bin
+
+# Perform static go tests.
+
+function usage {
+	echo "Usage $0 [OPTIONS] [PACKAGES]"
+	echo "Perform static go checks on PACKAGES (./... by default)."
+	echo
+	echo "List of options:"
+	echo "  -h, --help             print this help"
+	echo "  -n, --no-network       do not access the network"
+}
+
+for i in "$@"; do
+	case $i in
+		-h|--help)
+			usage
+			exit 0
+			;;
+		-n|--no-network)
+			NONETWORK=1
+			shift
+			;;
+		*)
+			args="$args $i"
+			;;
+	esac
+done
+
+go_packages=$args
+
+# process everything in the tree, apart from any vendor bits, as they are
+# not our responsibility to check and fix
+[ -z "$go_packages" ] && {
+	go_packages=$(go list ./... | grep -v vendor)
+}
+
+function install_package {
+	url="$1"
+	name=${url##*/}
+
+	if [ -n "$NONETWORK" ]; then
+		echo "Skipping updating package $name, no network access allowed"
+		return
+	fi
+
+	echo Updating $name...
+	go get -u $url
+}
+
+# Ensure the checkers are installed and upto date
+install_package github.com/fzipp/gocyclo
+install_package github.com/client9/misspell/cmd/misspell
+install_package github.com/golang/lint/golint
+install_package github.com/gordonklaus/ineffassign
+
+echo Doing go static checks on packages: $go_packages
+
+echo "Running misspell..."
+go list -f '{{.Dir}}/*.go' $go_packages |\
+	xargs -I % bash -c "misspell -error %"
+
+echo "Running go vet..."
+go vet $go_packages
+
+cmd="gofmt -s -d -l"
+echo "Running gofmt..."
+diff=$(find . -not -wholename '*/vendor/*' -name '*.go' | \
```

Comment:
It would be slightly more efficient to not walk the vendor tree, rather than walk it and filter out the names which contain vendor.

    find . \( -name vendor -prune \) -o -name '*.go'

Translation: If the name matches 'vendor' then prune off everything below it and move onto the next thing, or if the name matches '*.go' then print the path